src/Scenario/Processor.php

```diff
@@ -64,19 +64,26 @@ public function execute(array $scenarios): void
             implode(', ', $scenarios),
             $this->packageManager->getPrettyInfo()
         ));
+        $steps = [];
 
         try {
             $this->mergedScenarios = $this->merger->merge($scenarios);
             $steps = $this->mergedScenarios['steps'];
-
+        } catch (Throwable $exception) {
+            $this->handleException(
+                $exception,
+                sprintf('Unhandled error: [%d] %s', $exception->getCode(), $exception->getMessage())
+            );
+        }
+        try {
             array_walk($steps, function (StepInterface $step, string $name) {
                 $this->logger->debug('Running step: ' . $name);
 
                 $step->execute();
 
                 $this->logger->debug(sprintf('Step "%s" finished', $name));
             });
-        } catch (StepException $stepException) {
+        } catch (Throwable $stepException) {
             try {
```

Comment:
Let's move nested try/catch block to a new separate method as this method contains too many try/catch blocks and it becomes difficult to read.
```
        } catch (Throwable $stepException) {
            $this->runActions();
            $this->handleException($stepException);
        }
```

Reply:
done